lib/types.ts

```diff
@@ -385,7 +385,7 @@ type DeepModify<T> =
 /** Remove keys with `never` value from object type */
 export type NonNever<T extends {}> = Pick<T, { [K in keyof T]: T[K] extends never ? never : K }[keyof T]>;
 
-export type NonEmptyObject<T extends {}> = keyof T extends never ? never : T;
+export type NonEmptyObject<T extends Record<PropertyKey, any>> = keyof T extends never ? never : T;
```

Comment:
`PropertyKey` is not friendly for `keyofStringsOnly` option in tsconfig.json so we implemented our own

There's also a link attached to it, please have a look at it for more information – https://devblogs.microsoft.com/typescript/announcing-typescript-2-9-2/#support-for-symbols-and-numeric-literals-in-keyof-and-mapped-object-types

Also you can use `AnyRecord` for it as it's already defined in the file:

```suggestion
export type NonEmptyObject<T extends AnyRecord> = keyof T extends never ? never : T;
```

Reply:
I didn't know about this.
TIL :D
Went through the article, and it makes sense to have a separate `KeyofBase` type to not run into problems due to `keyofStringsOnly` compiler option.

And since we already have this `AnyRecord` implementation better to use it.